### PR TITLE
Add vanilla es6 react snippets

### PR DESCRIPTION
**Pull request by IgorGee**
This snippets file is an updated version of this repo:
https://github.com/justinj/vim-react-snippets

I have removed any snippets dealing with deprecated functions.

### CODE REVIEW

**Comment by lpil** [comment]
cool. :)

Could you change the file name to match the other javascript library snippets please?

**Comment by IgorGee** [comment]
No problem.

Should I move it into the javascript folder and rename it to something like 'javascript-react.snippets'?

**Comment by lpil** [comment]
Could you put the semicolons back please?

**Comment by IgorGee** [comment]
Somehow I knew that commit wasn't go to fly :/

**Comment by michaelghinrichs** [comment]
Looking at [this post](https://babeljs.io/blog/2015/06/07/react-on-es6-plus) it seems like some of these these are actually es5 syntax, not es6.

Can you update the values that are still being done in the es5 way? ->

For Props:
es5
```
 propTypes: {
    prop: React.PropTypes.type,
}
```

es6
```
static propTypes = {
   prop: React.PropTypes.type,
}
```

For Default props:
es5
```
getDefaultProps: function() {
    return {
       prop: 'value',
    };
  },
```

es6
```
  static defaultProps = {
     prop: 'value',
  }
```

File: snippets/javascript/javascript-react.snippets [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		${1}
+	}
+snippet cwu
+	componentWillUpdate(nextProps, nextState) {
+		${1}
+	}
+snippet fup
+	forceUpdate(${1:callback});
+snippet gdp
+	getDefaultProps() {
```
Comment (michaelghinrichs) [inline snippets/javascript/javascript-react.snippets]: convert to es6

File: snippets/javascript/javascript-react.snippets [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		};
+	}
+snippet gis
+	getInitialState() {
+		return {
+			${1}: ${2}
+		};
+	}
+snippet pt
+	propTypes: {
```
Comment (michaelghinrichs) [inline snippets/javascript/javascript-react.snippets]: convert to es6

**Comment by michaelghinrichs** [comment]
Awesome! thanks @IgorGee :) completely missed state!

**Comment by IgorGee** [comment]
Happy to contribute.
Thank you for showing me the post @michaelghinrichs.

**Pull request merged**